src/dso_hdr.cc

```diff
@@ -243,21 +243,36 @@ DDProfModRange DsoHdr::compute_mod_range(DsoMapConstIt it, const DsoMap &map) {
     return DDProfModRange();
   }
   DsoMapConstIt first_el = it;
-  while (first_el != map.begin()) {
-    --first_el;
-    if (it->second._filename != first_el->second._filename) {
-      ++first_el;
-      break;
+  {
```

Comment:
```  
  const ddprof::Dso *dso = nullptr;
  auto func = [&](auto &x) {
    if (x.second._filename == it->second._filename) {
      dso = &x.second;
      return false;
    };
    return x.second._type != dso::DsoType::kAnon;
  };
  std::find_if(it, std::end(map), func);
  uint64_t high_addr = dso->_end;
  std::find_if(std::make_reverse_iterator(std::next(it)), std::rend(map), func);
  uint64_t low_addr = dso->_start;
  return DDProfModRange{._low_addr = low_addr, ._high_addr = high_addr};
```
```suggestion
  const ddprof::Dso *dso = nullptr;
  auto func = [&](auto &x) {
    if (x.second._filename == it->second._filename) {
      dso = &x.second;
      return false;
    };
    return x.second._type != dso::DsoType::kAnon;
  };
  std::find_if(it, std::end(map), func);
  uint64_t high_addr = dso->_end;
  std::find_if(std::make_reverse_iterator(std::next(it)), std::rend(map), func);
  uint64_t low_addr = dso->_start;
  return DDProfModRange{._low_addr = low_addr, ._high_addr = high_addr};  
```

Reply:
Not sure if it's better ;)

Reply:
It is more compact and stylish ! Though it took me some time to read it :-)